docs/training/hyperparameter/search-methods/hp-custom.rst

```diff
@@ -0,0 +1,219 @@
+           with checkpoint_path.open("w") as f:
+               # populate a dictionary or another serializable data structure
+               # with the internal state data
+               # you can use any serialization format (not just json)
+               state = {}
+               json.dump(state, f)
+
+       def load_method_state(self, path: Path) -> None:
+           checkpoint_path = path.joinpath("method_state.json")
+           with checkpoint_path.open("r") as f:
+               state = json.load(f)
+               # initialize internal state from the deserialized data structure
+
+***********************************
+ Run Hyperparameter Search Locally
+***********************************
+
+A script performing hyperparameter tuning using :class:`~determined.searcher.LocalSearchRunner` may
+look like the following:
+
+.. code:: python
+
+   import logging
+   from pathlib import Path
+   from determined import searcher
+
+
+   if __name__ == "__main__":
+       # The content of the following directory is uploaded to Determined cluster.
+       # It should include all files necessary to run the experiment (as usual).
+       model_context_dir = "experiment_files"
+
+       # Path to the .yaml file with the multi-trial experiment configuration.
+       model_config = "experiment_files/config.yaml"
+
+       # While LocalSearchRunner saves its own state and ensures invoking save() and
+       # load() methods when necessary, a user is responsible for implementing
+       # SearchMethod.save_method_state() and SearchMethod.load_method_state() to ensure
+       # correct resumption of the SearchMethod.
+       searcher_dir = Path("local_search_runner/searcher_dir")
+
+       # instantiate your search method, passing the necessary parameters
+       search_method = MySearchMethod(...)
+
+       search_runner = searcher.LocalSearchRunner(search_method, searcher_dir=searcher_dir)
+
+       experiment_id = search_runner.run(model_config, model_dir=model_context_dir)
+       logging.info(f"Experiment {experiment_id} has been completed.")
+
+****************************************
+ Run Hyperparameter Search on a Cluster
+****************************************
+
+A script to run your custom search method on a Determined cluster may look like this:
+
+.. code:: python
+
+   import determined as det
+   from pathlib import Path
+   from determined import searcher
+
+   if __name__ == "__main__":
+       # The content of the following directory is uploaded to Determined cluster.
+       # It should include all files necessary to run the experiment (as usual).
+       model_context_dir = "experiment_files"
+
+       # Path to the .yaml file with the multi-trial experiment configuration.
+       model_config = "experiment_files/config.yaml"
+
+       with det.core.init() as core_context:
+
+           info = det.get_cluster_info()
+           assert info is not None
+           args = AttrDict(info.trial.hparams)
```

Comment:
I'd prefer we leave `AttrDict` out of this, because it's not really relevant to what's happening.  Also it's coming from a package that isn't imported in this script, and adding imports to the script would further distract from the purpose of the example.